### PR TITLE
Add support for Ruby 3.1 

### PR DESCRIPTION
**Pull request by IvanTakarlikov-st**
Adds support of new required syntax of psych gem. 

Part of [Ruby 3 upgrade](https://github.com/sensortower/orbital_command/issues/2931)

### CODE REVIEW

**Review by andreyzhelnin-st — APPROVED** [review]
LGTM

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -53,6 +62,15 @@ jobs:
           rails: '6.1'
           gemfile: gemfiles/rails-6.1.gemfile
           experimental: false
+        - mongodb: '4.4'
+          ruby: ruby-3.1
+          topology: replica_set
+          task: test
+          os: ubuntu-20.04
```
Comment (ST-eugenekrivdyuk) [inline .github/workflows/test.yml:69]: Is Ubuntu 22.04 supported?

**Review by ST-eugenekrivdyuk — APPROVED** [review]

File: .github/workflows/test.yml (hunk 1 above)
Comment (IvanTakarlikov-st) [reply .github/workflows/test.yml:69]: let's give it a try

Comment (IvanTakarlikov-st) [reply .github/workflows/test.yml:69]: Well, it doesn't work, even Mongo team [runs on Ubuntu 20.04](https://github.com/mongodb/mongoid/blob/master/.github/workflows/test.yml#L24), so, let's stick with this.

**Pull request merged**